### ISSUE TITLE
[BUG] NetworkManager configs should be readable by non-root user

### ISSUE THREAD

**Issue by siteshwar** [code-ready/crc#1152]
### General information

  * OS: Linux 
  * Hypervisor: KVM
  * Did you run `crc setup` before starting it (Yes/No)? Yes

## CRC version
```bash
crc version: 1.8.0+0a318dc
OpenShift version: 4.3.8 (embedded in binary)

```
  
## CRC status
```bash
CRC VM:          Stopped
OpenShift:       Stopped
Disk Usage:      0B of 0B (Inside the CRC VM)
Cache Usage:     11.8GB
Cache Directory: /home/situ/.crc/cache
```

## CRC config
```bash
```

## Host Operating System
```bash
NAME=Fedora
VERSION="31 (KDE Plasma)"
ID=fedora
VERSION_ID=31
VERSION_CODENAME=""
PLATFORM_ID="platform:f31"
PRETTY_NAME="Fedora 31 (KDE Plasma)"
ANSI_COLOR="0;34"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:31"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f31/system-administrators-guide/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=31
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=31
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="KDE Plasma"
VARIANT_ID=kde
```

### Steps to reproduce

  1. Download tar archive for linux from [here](https://cloud.redhat.com/openshift/install/crc/installer-provisioned)
  2. Run `crc setup` as a non-root user
  3. Run `crc start` as non-root user

### Expected
`crc start` should run succesfully.

### Actual
During setup I get these errors:

```
FATA Error opening file: /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf: open /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf: permission denied 
FATA Error opening file: /etc/NetworkManager/dnsmasq.d/crc.conf: open /etc/NetworkManager/dnsmasq.d/crc.conf: permission denied 
```

These files were only readable and writable by `root` user on my system. It may be just a convenience issue as I was able to fix it by changing permissions on the file, but it would be good if we can make it work by default.

**Comment by praveenkumar** [comment]
@siteshwar It should since we are updating these file using `sudo` user, didn't you get the prompt for sudo user password during the setup?

**Comment by siteshwar** [comment]
I got the prompt, but permissions on these files depend on umask. On my system, it's 

```
> sudo umask
[sudo] password for situ: 
0077
```

So these configuration files were not accessible to non-root users.

**Comment by praveenkumar** [comment]
> permissions on these files depend on umask

@siteshwar This is not the default settings iirc (at least on Gnome) by any chance you set this permission to those files ?

**Comment by siteshwar** [comment]
I just ran `crc setup` and it set permissions on these files.

**Comment by praveenkumar** [comment]
> I just ran crc setup and it set permissions on these files.

@siteshwar the files/dir created using the `sudo` but since your system umask is `0077` which means it has severe security level so any other user not able to read the file and looks like we don't explicit set the file mode :( https://www.cyberciti.biz/tips/understanding-linux-unix-umask-value-usage.html

**Comment by praveenkumar** [comment]
fixed by #1158
